nexus/db-queries/src/db/queries/virtual_provisioning_collection_update.rs

```diff
@@ -246,15 +238,7 @@ WITH
     ),")
                 .bind::<sql_types::Uuid, _>(id)
             },
-            UpdateKind::DeleteInstance { id, max_instance_gen, .. } => {
-                // The filter condition here ensures that the provisioning record is
-                // only deleted if the corresponding instance has a generation
-                // number less than the supplied `max_instance_gen`. This allows a
-                // caller that is about to apply an instance update that will stop
-                // the instance and that bears generation G to avoid deleting
-                // resources if the instance generation was already advanced to or
-                // past G.
```

Comment:
This is no longer necessary because only the update saga does this, and it has its own mutual exclusion mechanism.